packages/core/src/dispatcher.ts

```diff
@@ -98,17 +99,9 @@ export class Dispatcher {
    */
   async retrieveCommit(cid: CID | string, streamId?: StreamID): Promise<any> {
     try {
-      const asCid = typeof cid === 'string' ? new CID(cid) : cid
-
-      // Lookup CID in cache before looking it up IPFS
-      const cachedDagNode = await this.dagNodeCache.get(asCid.toString())
-      if (cachedDagNode) return cloneDeep(cachedDagNode)
-
-      // Now lookup CID in IPFS and also store it in the cache
-      const dagNode = await this._ipfs.dag.get(asCid, { timeout: IPFS_GET_TIMEOUT })
+      const result = await this._getFromIpfs(cid)
       await this._restrictCommitSize(cid)
```

Comment:
Can this IPFS API call also fail? If so, should it also be retried? Should we have generic retry handling that gets commonly applied across all IPFS API calls?

Reply:
Yes in theory, but since if we get this far that means we've already loaded the CID into our local IPFS node, it should be extremely unlikely for this to fail